Update to fix #62377, #64045

krisvasudevan opened the pull request:
##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
Fixes #62377, #64045
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Bugfix Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
nios_a_record.py
nios_txt_record.py
api.py

##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->
Added new parameters to A record and TXT record to handle modification of content in the records.
In case of A record, adding this new parameter enables modifying the IP address of a specific record when there are multiple records with the same name.
Similarly for TXT record, adding the new parameters enable modifying the text content of a specific TXT record when there are multiple records with the same name.
<!--- Paste verbatim command output below, e.g. before and after your change -->
```paste below

```

krisvasudevan committed:
Update to fix #62377, #64045

Akasurde reviewed (changes requested):
@krisvasudevan Thanks for the contribution. Could you please update the title of PR and description of PR to reflect what is being changed? Also, add a changelog entry [here](https://github.com/ansible/ansible/tree/devel/changelogs/fragments) so that we can backport this change?

Thanks.

krisvasudevan commented:
My understanding from @justjais is that these changes cannot be backported since I have added new parameters. The bugs I fixed (#62377, #64045) needed new parameters to be added to the nios_a_record and nios_txt_record modules.

justjais commented on lib/ansible/modules/net_tools/nios/nios_a_record.py:
```diff
@@ -116,6 +116,17 @@
+- name: update an A record IP address
+  nios_a_record:
+    name: a.ansible.com
+    ipv4: {new_ipv4addr: 10.10.10.1, old_ipv4addr: 192.168.10.1}
```
Plz comment the new param added in module doc (ref: [nios_host_record](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/net_tools/nios/nios_host_record.py#L31)).

krisvasudevan replied:
Done for both nios_a_record and nios_txt_record

justjais commented:
@krisvasudevan yes, this can't be backported as new params are introduced, and can you update the PR fixing the shippable failure. 
Also, can you update the PR description accordingly.

krisvasudevan committed:
Documentation fixes in nios modules

Update nios_a_record.py and nios_txt_record.py

krisvasudevan commented:
> @krisvasudevan yes, this can't be backported as new params are introduced, and can you update the PR fixing the shippable failure.
> Also, can you update the PR description accordingly.
Could you please give me some clarity on this?

justjais commented on lib/ansible/modules/net_tools/nios/nios_a_record.py:
```diff
@@ -27,7 +27,8 @@
       - Specifies the fully qualified hostname to add or remove from
-        the system
+        the system. User can also update the hostname as it is possible
+        to pass a dict containing I(new_name), I(old_name). See examples.
```
```suggestion
        to pass a dict containing I(new_ipv4addr), I(new_ipv4addr). See examples.
```

justjais reviewed (changes requested):
Also, can u update the issue description with appropriate issue description as currently it mentions only the open issue IDs.

krisvasudevan committed:
Update lib/ansible/modules/net_tools/nios/nios_a_record.py

Co-Authored-By: Sumit Jaiswal <sjaiswal@redhat.com>

krisvasudevan commented:
I've added information in the Additional Information section.

sugitk commented:
@krisamundson 

Could you please update the status of this PR?  The customer is still waiting for the fix.
Thank you.

danielmellado commented:
Thank you very much for your interest in Ansible. This plugin is no longer maintained in this repository and has been migrated to https://github.com/ansible-collections/community.general
Please re-submit this issue in the above repository.
If you have further questions please stop by IRC or the mailing list:

IRC: #ansible-network on irc.freenode.net
mailing list: https://groups.google.com/forum/#!forum/ansible-project
